### PR TITLE
Task1

### PR DESCRIPTION
**Pull request by sudarsankonda**
Angular task1 completed as per homework guideline. Please review

### CODE REVIEW

File: src/app/components/edit-article/edit-article.component.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+      title: ['', Validators.required],
+      description: ['', Validators.required],
+      content: ['', Validators.required],
+      imageUrl: ['', Validators.required],
+      datePosted: ['', Validators.required],
+      author: '',
+      sourceUrl: ''
+    });
+
+    if (this.activatedRoute.snapshot.paramMap.get('id')) {
```
[the remark this answers is not in the document]
Comment (st-zelenin) [reply src/app/components/edit-article/edit-article.component.ts]: UPDATED: now I see that you use `edit-article` from for create new as well.. Do you agree that it is a little bit unexpected to have `edit-article` for `create new` functionality? For such cases I suggest to split components:
- `article-form` component - the base/shared component with form controls (@'Input is optional, @'Output - the form submit callback)
- `new-article` component - on `article-form` submit emits `AddArticle` action.
- `edit-article` component - on load fetches Article by `id` (or fins it in the store, which is not correct), on `article-form` submit emits `EditArticle` action (or makes API call).

File: src/app/components/edit-article/edit-article.component.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      this.isEdit = true;
+      this.id = parseInt(this.activatedRoute.snapshot.paramMap.get('id'));
+      this.heading = 'Edit';
+      this.buttonText = 'Update';
+      this.storeSubscription = this.store.select(selectArticles).subscribe(articles => {
```
[the remark this answers is not in the document]
Comment (st-zelenin) [reply src/app/components/edit-article/edit-article.component.ts]: UPDATED: I do not think that it was a good idea to use store for this home task.. Please think if store will be convenient for the second home task.

File: src/app/components/edit-article/edit-article.component.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  }
+
+  ngOnDestroy() {
+    if (this.isEdit) {
+      this.storeSubscription.unsubscribe();
+    }
+  }
+
+  onSubmit() {
+    if (this.isEdit) {
```
[the remark this answers is not in the document]
Comment (st-zelenin) [reply src/app/components/edit-article/edit-article.component.ts]: UPDATED: I got the idea. Please see my recommendation above.

File: src/app/components/list-articles/list-articles.component.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                    article.content.toLowerCase().indexOf(searchText.toLowerCase()) !== -1);
+  }
+
+  onCreatedByMe() {
+    this.articles = this.createdByMe ?
+            this.storeArticles.filter(article => article.source.name === 'Sudarsan') :
+            this.storeArticles;
+  }
+
+  createForm() {
```
Comment (st-zelenin) [inline src/app/components/list-articles/list-articles.component.ts]: why is it empty?

**Comment by sudarsankonda** [comment]
merge to master as PR approved by Stepan

**Pull request merged**